test/fakes/fakeservercatalog/catalog.go

```diff
@@ -45,3 +47,27 @@ func (c *Catalog) SetUpstreamCA(upstreamCA upstreamca.UpstreamCA) {
 func (c *Catalog) SetKeyManager(keyManager keymanager.KeyManager) {
 	c.KeyManager = keyManager
 }
+
+func (c *Catalog) AddNotifier(notifier catalog.Notifier) {
+	c.Notifiers = append(c.Notifiers, notifier)
```

Comment:
the catalog itself isn't protected by a mutex since it isn't expected to change once loaded. in a similar manner, these methods are used to modify the catalog before you run code that uses it. it isn't perfect but there is no race that i know of in practice.